client/index.js

```diff
@@ -35,7 +36,11 @@ let lastAppProps
 export const router = createRouter(pathname, query, getURL(), {
   Component,
   ErrorComponent,
-  err
+  err,
+  formatURL: exported && function (buildId, route) {
+    route = route && route.replace(/\/$/, '')
+    return route + '/index.json'
```

Comment:
Maybe I'm missing something but if `route` is falsy then we are returning incorrect routes like `undefined/index.js`